### PR TITLE
Fixes u'' string in superfluous-parens message

### PR DESCRIPTION
**Pull request by kczapla**

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -4,6 +4,10 @@ Pylint's ChangeLog
 What's New in Pylint 1.8?
 =========================
 
+	* Fixing u'' string in superfluous-parens message
```
Comment (rogalski) [inline ChangeLog]: please use spaces exclusively; also we need similar entry in whats_new/1.8.rst file

File: pylint/utils.py [hunk 2]
```diff
@@ -398,6 +398,8 @@ def add_message(self, msg_descr, line=None, node=None, args=None, confidence=UND
         # expand message ?
         msg = msg_info.msg
         if args:
+            if isinstance(args, six.text_type):
```
Comment (rogalski) [inline pylint/utils.py]: please add failing unit or functional test

**Comment by kczapla** [comment]
@rogalski made an updates u requested.

File: pylint/test/unittest_checker_format.py [hunk 3]
```diff
@@ -138,6 +138,20 @@ def testCheckKeywordParensHandlesUnnecessaryParens(self):
             with self.assertAddsMessages(msg):
                 self.checker._check_keyword_parentheses(_tokenize_str(code), offset)
 
+    def testCheckIfArgsAreUnicode(self):
+        self.checker._keywords_with_parens = set()
+        cases = [
+            (Message('superfluous-parens', line=1, args='if'),
+             'if (foo):', 0),
+            (Message('superfluous-parens', line=1, args='asssert'),
+             'assert (1 == 1)', 0)
```
Comment (rogalski) [inline pylint/test/unittest_checker_format.py]: Shoudn't we test unicode-encoded code string here (`u'if (foo):'`)? From what I see those new tests will pass even without your patch.

Comment (kczapla) [reply pylint/test/unittest_checker_format.py]: You are right. I lost somewhere u'' during the final edition.

**Comment by rogalski** [comment]
Thank you!

File: pylint/test/unittest_checker_format.py [hunk 4]
```diff
@@ -138,6 +138,20 @@ def testCheckKeywordParensHandlesUnnecessaryParens(self):
             with self.assertAddsMessages(msg):
                 self.checker._check_keyword_parentheses(_tokenize_str(code), offset)
 
+    def testCheckIfArgsAreNotUnicode(self):
+        self.checker._keywords_with_parens = set()
+        cases = [
+            (Message('superfluous-parens', line=1, args='if'),
```
Comment (rogalski) [inline pylint/test/unittest_checker_format.py]: Sorry to mess once again with this change, but `Message()` instances are unused, they should be removed before merge.

**Pull request merged**